lib/StaticAnalyzer/Core/PlistDiagnostics.cpp

```diff
@@ -119,6 +119,26 @@ static void EmitMessage(raw_ostream &o, StringRef Message, unsigned indent) {
   EmitString(o, Message) << '\n';
 }
 
+static void EmitConstraints(raw_ostream &o,
+                            const InternalInfoMap &Map,
+                            unsigned indent) {
+  if (Map.empty())
+    return;
+
+  Indent(o, indent) << "<key>internalinfos</key>\n";
+  Indent(o, indent) << "<array>\n";
+  ++indent;
+  for (const auto &Info : Map) {
+    ++indent;
```

Comment:
Could use a RAII object to manage `indent`.

Reply:
ctor: ++indent, dtor: --indent

Reply:
Actually, the entire file is managed like this. It would take a tremendous effort to refactor the whole file -- I'd rather keep consistency for now.